cadcutils/cadcutils/util/utils.py

```diff
@@ -172,43 +173,155 @@ def parse_resource_id(resource_id):
+    the common options are automatically added to the base parser.
+
+    :param subparsers: True if the parser will use subparsers (subcommands) otherwise False
     :param version: A version number if desired.
-    :param usecert: If True add '--certfile' argument.
+    :param usecert: If True add '--cert' argument.
     :param default_resource_id: default resource identifier to use
     :return: An ArgumentParser instance.
     """
-    parser = ArgumentParser(add_help=False)
+    cparser = ArgumentParser(add_help=False, formatter_class=SingleMetavarHelpFormatter)
+
+    if version is not None:
+        cparser.add_argument('-V', '--version', action='version', version=version)
+    auth_group = cparser.add_mutually_exclusive_group()
     if usecert:
-        parser.add_argument('--certfile', type=str,
-                            help="location of your CADC certificate "
-                            + "file (default: $HOME/.ssl/cadcproxy.pem, " +
-                            "otherwise uses $HOME/.netrc for name/password)",
-                            default=os.path.join(os.getenv("HOME", "."),
-                                                 ".ssl/cadcproxy.pem"))
-    parser.add_argument('--anonymous', action="store_true",
-                        help='Force anonymous connection')
-    parser.add_argument('--host', help="Base hostname for services - used mainly for testing " +
-                                       "(default: www.cadc-ccda.hia-iha.nrc-cnrc.gc.ca)",
+        auth_group.add_argument('--cert', type=str,
+                            help='location of your X509 certificate to use for authentication ' +
+                            '(unencrypted, in PEM format)')
+    auth_group.add_argument('-n', action='store_true', help='Use .netrc in $HOME for authentication')
+    auth_group.add_argument('--netrc-file', help='netrc file to use for authentication')
+    auth_group.add_argument('-u', '--user', help='Name of user to authenticate. ' +
+                             'Note: application prompts for the corresponding password!')
+    cparser.add_argument('--host',
+                        help='Base hostname for services - used mainly for testing ' +
+                                       '(default: www.cadc-ccda.hia-iha.nrc-cnrc.gc.ca)',
                         default='www.cadc-ccda.hia-iha.nrc-cnrc.gc.ca')
     if default_resource_id is None:
-        parser.add_argument('--resourceID', type=urlparse, required=True,
-                            help="resource identifier (e.g. ivo://cadc.nrc.ca/caom2repo")
+        cparser.add_argument('--resourceID',
+                            type=urlparse, required=True,
+                            help='resource identifier (e.g. ivo://cadc.nrc.ca/service')
     else:
-        parser.add_argument('--resourceID', type=parse_resource_id,
+        cparser.add_argument('--resourceID',type=parse_resource_id,
```

Comment:
in an application like 'get_data' I hope we are not requiring that the resourceID be give via a flag option.  i.e.:

cadc-get-data ad:CFHT/XXXXXXX   not

cadc-get-data --resourceID ad:CFHT/XXXXXX